pkg/ottl/ottlfuncs/func_uuid.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func UUID[K any]() (ottl.ExprFunc[K], error) {
+	return func(ctx context.Context, tCtx K) (interface{}, error) {
+		u := uuid.New()
+		return u.String(), nil
```

Comment:
I'm curious what you think about this returning a []byte instead of a string.  If it returns a []byte then users could do generate new trace ids quickly: `set(trace_id, TraceID(Uuid())`

Might be worth a configuration parameter that lets you select the format of your UUID. 

@evan-bradley @kentquirk @bogdandrutu

Reply:
Not sure, the user request this came from was specifically to set the uuid as a log attribute.

Reply:
This is now returning a string, so we're covered. Closing the thread.

Reply:
@atoulme returning only a string is good, but what are your thoughts on @kentquirk request for a parameter specifying version number?

Getting the parameters right at the start is important to prevent breaking changes in the future.

Reply:
Missed that sorry.


I don't really have strong opinions on this, except I'm not sure that should be a parameter. UUID v4 should be default unless the machine lacks entropy as well imo.
We could create the `UUID()` function and alias it as `UUIDv4()` , and then create down the road additional `UUIDvX()` functions.

Reply:
Looking at how Go handles this topic they create different functions for non-v4 versions (I believe) and leave New() for fully random UUIDs. With that in mind, I think we're good for this function to mimic that pattern.